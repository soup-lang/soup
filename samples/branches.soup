func f(a, b, cb)
    setTimeout(%, 1)
    cb(a * a + b * b, a * a - b * b)

f(0, 1, %(x, y))

if x > y
    setTimeout(%, 1)
    m: x + y
    n: x - y
    console.log(m, n)

    f(m, n, %(p, q))
    console.log(p, q)

m: 10
n: 42

f(m, n, %(p, q))

if p > q
    m: p + q
    n: p - q
    console.log(m, n)

console.log(m, n)

if p > q
    f(10, 42, %(m, n))
    console.log(m, n)

console.log(m, n)
