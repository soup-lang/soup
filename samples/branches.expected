0 2
4 -4
200 3528
10 42
1864 -1664
10 42
